qcodes/dataset/data_set.py

```diff
@@ -637,7 +641,7 @@ def load_by_counter(counter, exp_id):
     return d
 
 
-def new_data_set(name, exp_id: Optional[int] = None,
+def new_data_set(name, exp_id: Optional[int] = None, conn=None,
```

Comment:
I guess any new keyword argument as a good rule of thumb should be added after the already existing ones.